helix-core/src/main/java/org/apache/helix/controller/dataproviders/ResourceControllerDataProvider.java

```diff
@@ -262,6 +265,14 @@ public void setCachedIdealMapping(String resource, ZNRecord mapping) {
     _idealMappingCache.put(resource, mapping);
   }
 
+  /**
+   * Return the set of PropertyTypes that changed prior to this round of rebalance.
+   * @return
+   */
+  public Set<HelixConstants.ChangeType> getRefreshedChangeTypes() {
+    return _refreshedChangeTypes;
```

Comment:
Init the set, or it will return null before any refresh.

Reply:
Moreover, if we are going to modify this set during the ResourceControllerDataProvider.refresh method, we will need to "synchronized" the method.

Reply:
It shouldn't be modifying. This should have been read-only. I'll return an unmodifiable set.

Reply:
Added `clearRefreshedChangeTypes()` in `ResourceControllerDataProvider`